staging/src/k8s.io/apiserver/pkg/util/compatibility/version.go

```diff
@@ -26,6 +26,18 @@ import (
 // so the emulation version cannot go lower than that.
 var minimumKubeEmulationVersion *version.Version = version.MajorMinor(1, 31)
 
+// KubeComponentEffectiveVersion returns the effective version of the kube component.
+// If the kube component is registered in the DefaultComponentGlobalsRegistry, the
+// effective version from the registry is returned. Otherwise, the DefaultBuildEffectiveVersion
+// is returned.
+func KubeComponentEffectiveVersion() basecompatibility.EffectiveVersion {
```

Comment:
Since this is a global, it is possible to call this *before* the parsing / initialization of the effective version has been done, right?

We tried to add tests into the cel initialization paths to detect when things initialized based on versions or gates prematurely, but moving this into a global used by things other than cel worries me a bit. How can we protect against accidentally using this in an initialization path too early?

Reply:
I've modified this so that the caller must provide a registry.  This makes it the callers responsibility to only pass in a registry that is initialized / ready for use.